packages/react-reconciler/src/ReactFiberUnwindWork.js

```diff
@@ -331,6 +353,41 @@ function throwException(
         // whole tree.
         renderDidSuspend(root, absoluteTimeoutMs, renderExpirationTime);
 
+        workInProgress.effectTag |= ShouldCapture;
+        workInProgress.expirationTime = renderExpirationTime;
+        return;
+      } else if (
+        enableSuspenseServerRenderer &&
+        workInProgress.tag === DehydratedSuspenseComponent &&
+        shouldCaptureDehydratedSuspense(workInProgress)
+      ) {
```

Comment:
Since this branch doesn't suspend (it doesn't call `renderDidSuspend`) I would expect React to keep rendering the same level over and over until the promise resolves. Is that what's happening?

Reply:
No. What happens is that only the first path schedules remaining work at "Never" expiration time. Then if it throws, it doesn't suspend but it also doesn't leave any work on it. Instead it commits. Then it waits for the retry. The retry gets scheduled at normal priority. If that update also throws a promise, then it commits in the dehydrated state again and waits for the retry.

Reply:
I see so when something suspends inside a dehydrated Suspense boundary it always bails out and clears the expiration time. The ping/retry adds the expiration time back. There’s no need to suspend the commit because it’s not blocking anything.